pkg/client/s3client.go

```diff
@@ -153,18 +154,22 @@ func (c *S3Client) CheckBucketLocationConstraint(bucketName string, bucketLocati
 	return false, errors.New("bucket location constraint doesn't match")
 }
 
-func (c *S3Client) CheckIfObjectExists(bucketName, objectName string) bool {
+func (c *S3Client) CheckIfObjectExists(bucketName, objectName string) (bool, error) {
 	input := &s3.GetObjectInput{
 		Bucket: aws.String(bucketName),
 		Key:    aws.String(objectName),
 	}
 
 	_, err := c.S3Session.GetObject(input)
+
 	if err != nil {
-		klog.Errorf("ERROR: %v", err)
-		return false
+		if strings.Contains(err.Error(), "NoSuchKey") {
```

Comment:
Instead of string compare, wondering if this cos-sdk exposes any errors so that we can directly compare with the error.